src/services/redisSessionStorage.ts

```diff
@@ -975,18 +980,15 @@ export default class RedisSessionStorage extends RedisStorageUtils
       // Delete all active tokens that are different
       // from the new one generated and provided inside user object.
       const deleteOldKeysResponseV2 = await pipe(
-        TE.of([...oldSessionInfoKeys, ...oldSessionKeys, ...externalTokens]),
-        TE.chain(
-          TE.fromPredicate(
-            keys => keys.length === 0,
-            () => true
-          )
-        ),
-        TE.foldW(TE.right, keys =>
-          pipe(
-            TE.tryCatch(() => this.redisClient.del(keys), E.toError),
-            this.integerReplyAsync()
-          )
+        [...oldSessionInfoKeys, ...oldSessionKeys, ...externalTokens],
+        TE.fromPredicate(keys => keys.length === 0, identity),
+        TE.fold(
```

Comment:
this fold was not correct looking at the old implementation logic. I proceeded to repair it and refactor this block of lines